backend/server.go

```diff
@@ -60,34 +62,40 @@ func WrapApiEndpoint(
 	return func(responseWriter http.ResponseWriter, request *http.Request) {
 		allowCors(responseWriter, request)
 
-		var httpStatus int
-		if err, httpStatus := apiHandlerFunc(responseWriter, request); err != nil {
-			if httpStatus >= 500 {
+		body, status, err := apiHandlerFunc(responseWriter, request)
+		if err != nil {
+			responseWriter.WriteHeader(status)
+			if status >= 500 {
 				log.Print(err)
 			}
-			json.NewEncoder(responseWriter).Encode(map[string]string{
-				"error": err.Error(),
-			})
-			responseWriter.WriteHeader(httpStatus)
+
+			if body == nil {
+				body = map[string]interface{}{
+					"error": err.Error(),
```

Comment:
Can we take body["error"]: err.Error() out of this if statement and put it in an if statement where err!= nil?

Reply:
this is already in an err != nil statement. Line 66